tests/components/numato/test_numato.py

```diff
@@ -0,0 +1,325 @@
+"""Tests for the numato integration."""
```

Comment:
Please split the tests between different modules, one per platform, eg `test_binary_sensor.py` and one more for the init module, `test_init.py`.

Reply:
Nearly done. @MartinHjelmare there is only one [failing test](https://github.com/home-assistant/core/pull/33816/files#diff-279d11a8e84ee23206d761455fe63af0R26) and I tried for hours to understand why it fails.
[Output](https://hastebin.com/raw/hifasotoli.log) indicates in line 108 that the state of `binary_sensor.numato_binary_sensor_mock_port2` is correctly set to off, but the [assertion](https://github.com/home-assistant/core/pull/33816/files#diff-279d11a8e84ee23206d761455fe63af0R37) fails anyways. I'm sort of desparate. Maybe it's something about async which I did wrong? Can you help?

Reply:
The log looks weird, ie it looks like it should work. Don't have a clue yet. I'll look more tomorrow.

Reply:
Thanks, @MartinHjelmare, I'd really appreciate it!

Reply:
I checked why this testcase worked before ripping it apart. It worked because of the service calls that were made for testing the switch. They all create a new task with hass.async_create_task under the hood.
So when I do the same and [create a dummy task](https://github.com/home-assistant/core/pull/33816/commits/d915bd9269c2a87ecae98373648ad1d163cbda1c#diff-279d11a8e84ee23206d761455fe63af0R49) it works.
@MartinHjelmare, maybe you get an idea now, what happens? I'd really want to remove the workaround!